scripts/gen-go-sdk.sh

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+# This script regenerates the Go SDK for a given HCP service ($1). Pass 'cloud-shared' to regenerate shared models.
+
+# The steps are:
+# 1. Remove the original SDK files for the service if they exist.
```

Comment:
Nit: If possible, I'd consider removing the original SDK files later in the process, maybe even right before writing the new files, so that a failure during the generation does not leave the SDK files missing.